R/backports.R

```diff
@@ -15,3 +15,13 @@ if (is.null(.BaseNamespaceEnv$lengths)) {
   }
 
 }
+
+if (is.null(.BaseNamespaceEnv$startsWith)) {
+
+  startsWith <- function(x, prefix) {
+    pattern <- sprintf("^\\Q%s\\E", prefix)
+    grepl(pattern, x, perl = TRUE)
+  }
+
+}
```

Comment:
Very separate from this PR, but it sparked my curiosity: I can see that renv doesn't have any lower bound R version constraint. For most packages I've worked on they've adopted something like the tidyverse's four versions which would mean we wouldn't need to keep this around (IIRC it was released in 3.3). 

But maybe the goal for renv is to be compatible back to very very early versions of R? I could see that being helpful for using renv to help modernize old projects. Is that right?